calledudeBot/Bots/TwitchBot.cs

```diff
@@ -1,90 +1,74 @@
 ﻿using calledudeBot.Chat;
 using calledudeBot.Services;
-using System;
 using System.Collections.Generic;
-using System.Globalization;
 using System.Linq;
 using System.Threading.Tasks;
 using System.Timers;
+using Nito.AsyncEx;
+using System;
 
 namespace calledudeBot.Bots
 {
     public sealed class TwitchBot : IrcClient
     {
-        private readonly RelayHandler messageHandler;
-        private List<string> mods = new List<string>();
+        private List<string> mods;
+        private OsuUserService osuUserService;
+        private RelayHandler messageHandler;
         private Timer modLockTimer;
         private bool modCheckLock;
-        private OsuUser oldOsuData;
-        private APIHandler<OsuUser> api;
-        private readonly string osuAPIToken, osuNick;
+        private readonly AsyncAutoResetEvent modWait = new AsyncAutoResetEvent();
+        private readonly string _osuAPIToken, _osuNick, _botNick;
+        private readonly OsuBot _osuBot;
 
-        public TwitchBot(string token, string osuAPIToken, string osuNick, string botNick, string channelName)
+        public TwitchBot(string token, string osuAPIToken, string osuNick, string botNick, string channelName, OsuBot osuBot)
             : base("irc.chat.twitch.tv", "Twitch", 366)
         {
             Token = token;
-            this.osuAPIToken = osuAPIToken;
-            this.osuNick = osuNick;
             this.channelName = channelName;
-
             nick = botNick;
-            messageHandler = new RelayHandler(this, channelName, osuAPIToken);
-            OnReady += onReady;
+
+            _osuAPIToken = osuAPIToken;
+            _osuNick = osuNick;
+            _botNick = botNick;
+            _osuBot = osuBot;
+
+            Ready += OnReady;
+            MessageReceived += HandleMessage;
+            UnhandledMessage += HandleRawMessage;
         }
 
-        private async Task onReady()
+        private async Task OnReady()
         {
             modLockTimer = new Timer(60000);
+            messageHandler = new RelayHandler(this, channelName, _osuAPIToken, _osuBot);
+            osuUserService = new OsuUserService(_osuAPIToken, _osuNick, this);
             modLockTimer.Elapsed += modLockEvent;
             modLockTimer.Start();
-            await WriteLine("CAP REQ :twitch.tv/commands");
 
-            GetMods();
-            api = new APIHandler<OsuUser>($"https://osu.ppy.sh/api/get_user?k={osuAPIToken}&u={osuNick}");
-            api.DataReceived += checkUserUpdate;
-            await api.Start();
+            await WriteLine("CAP REQ :twitch.tv/commands");
+            await osuUserService.Start();
         }
 
-        protected override async Task Listen()
+        private async void HandleMessage(string message, string user)
```

Comment:
Why do you sometimes use PascalCase for private methods and sometimes use camelCase?

Reply:
Basically the same as the other answer, I recently switched/decided to be more consistent to existing conventions, so my codebase looks like this in places where the code is older